epyqlib/devicetreeview.py

```diff
@@ -112,6 +112,13 @@ def context_menu(self, position):
             remove_device_action = menu.addAction("Close")
         if isinstance(node, epyqlib.devicetree.Bus):
             add_device_action = menu.addAction("Open device file...")
+            change_device_number = menu.addAction("Change device number...")
+            change_device_number.setEnabled(
+                not node._checked.name
+                and not node.fields.name == "Offline"
+                and node.interface == "pcan"
+                and node.bus.bus is None
+            )
```

Comment:
Does this only work when the particular adapter is not in use?  That's in regards to the `not node._checked.name` and `node.bus.bus is None` bits.  They do also seem a bit redundant theoretically?

The offline and pcan checks seem perhaps redundant as well.

Of course, sometimes redundancy is good...  I'm fine either way as long as either you have considered this or my misunderstandings are corrected.

Reply:
I followed the behavior of the `Load firmware...` and `Blink adapter` options (just below). If there is overlap between the tests, it wasn't apparent to me, but is certainly possible. You also ask if it is possible to change the device number while the adapter is active. My guess is, yes for receive active, since it seems to work with PCAN-View. It's unclear how it would behave with transmit active, but I could run some tests. That said, I'm guessing that the use case would be for a user to change the ID at the very beginning prior to selecting the device file anyways? And the use case of changing while the device is actively rx/tx would be unlikely. Correct me if I'm wrong.

Reply:
Usually you would set the device id once when you are physically setting them up.  Or if you have three you would set the IDs when you get them and each would be unique for you wherever you plugged them in.  So yeah, it's not a commonly done activity (I don't think...).  I would think this feature add is primarily justified by the read and presentation side of things.  Set is just a side note to be added while in all that code already.  Also, as to setting while operating, I don't know of this number doing \_anything\_ other than getting read and set.  I wouldn't guess it would be a problem to change while running.  If you want to test and change, go for it.  Whatever.

I'll accept consistency of the checks as sufficient for now.  `:]`